### PR TITLE
Console color only if JDK 22 says isConsole

### PR DESCRIPTION
**Pull request by som-snytt**
Reverted in https://github.com/scala/scala/pull/10758

Color output also depends on isConsole on jdk 22

### CODE REVIEW

**Comment by SethTisue** [comment]
@som-snytt thank you! mind submitting a 2.12.x backport?

**Comment by som-snytt** [comment]
@SethTisue did it work? your wish or even idle supposition is my command.

**Comment by SethTisue** [comment]
@som-snytt oh we're not done yet, there are still test failures at https://github.com/scala/scala/actions/runs/8665032593

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala [hunk 1]
```diff
@@ -59,7 +60,15 @@ trait ShellConfig {
   def batchText: String
   def batchMode: Boolean
   def doCompletion: Boolean
-  def haveInteractiveConsole: Boolean
+  def haveInteractiveConsole: Boolean = System.console != null && consoleIsTerminal
+
+  // false if JDK 22 and the system console says !isTerminal
+  def consoleIsTerminal: Boolean = {
+    def isTerminal: Boolean =
+      try classOf[java.io.Console].getMethod("isTerminal", null).invoke(System.console).asInstanceOf[Boolean]
```
Comment (lrytz) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala:68]: On my macOS

```
➜ sandbox java -version
openjdk version "22" 2024-03-19
OpenJDK Runtime Environment Temurin-22+36 (build 22+36)
OpenJDK 64-Bit Server VM Temurin-22+36 (build 22+36, mixed mode)
➜ sandbox s
> scala repl -S 2.13 -cp .
Welcome to Scala 2.13.13 (OpenJDK 64-Bit Server VM, Java 22).
Type in expressions for evaluation. Or try :help.

scala> classOf[java.io.Console].getMethod("isTerminal", null)
java.lang.NoSuchMethodException: java.io.Console.isTerminal(null)
  at java.base/java.lang.Class.getMethod(Class.java:2405)
  ... 30 elided
```

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala [hunk 2]
```diff
@@ -38,7 +39,7 @@ object ShellConfig {
       val batchText: String = if (settings.execute.isSetByUser) settings.execute.value else ""
       val batchMode: Boolean = batchText.nonEmpty
       val doCompletion: Boolean = !(settings.noCompletion.value || batchMode)
-      val haveInteractiveConsole: Boolean = !settings.Xnojline.value
+      override val haveInteractiveConsole: Boolean = super.haveInteractiveConsole && !settings.Xnojline.value
```
Comment (lrytz) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala:42]: This broke JLine support for me - it needs to be a lazy val due to safe init.

**Pull request merged**